spotlight-client/src/stores/RootStore.ts

```diff
@@ -0,0 +1,53 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import { Auth0ClientOptions } from "@auth0/auth0-spa-js";
+import UserStore from "./UserStore";
+
+/**
+ * Returns the auth settings configured for the current environment, if any.
+ */
+export function getAuthSettings(): Auth0ClientOptions | undefined {
+  // NOTE: there is no production auth requirement!
+  if (process.env.REACT_APP_AUTH_ENV === "development") {
+    return {
+      domain: "spotlight-login-staging.recidiviz.org",
+      client_id: "ID9plpd8j4vaUin9rPTGxWlJoknSkDX1",
+      redirect_uri: `${window.location.protocol}//${window.location.host}`,
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Returns the status of the auth requirement flag for the current environment.
+ */
+export function isAuthEnabled(): boolean {
+  return process.env.REACT_APP_AUTH_ENABLED === "true";
+}
+
+export default class RootStore {
+  userStore: UserStore;
+
+  constructor() {
+    this.userStore = new UserStore({
+      authSettings: getAuthSettings(),
```

Comment:
Why did you decide to pass in `authSettings` and `isAuthEnabled` here instead of keeping all of the auth code in the `UserStore`? For easier testing?

Reply:
not primarily for that reason, though it did improve the tests as a bonus ... I didn't want to directly couple the UserStore class to the environment, I wanted it to stay as self-contained as possible. The RootStore, being at the top of the tree, seemed like a more appropriate place to be interacting directly with the environment to let its children remain more modular.